### PR TITLE
Implement galleri

### PR DESCRIPTION
**Pull request by Arashfa0301**
Implemented galleri tab that previews images from last years itDAGENE. The pictures are placed in infra cdn folder in the server and are imported there from. 
-------
before:

image

After:

image

 
after:

I have tested both mobile and window view of this pr thoroughly.

### CODE REVIEW

File: components/Header/index.tsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ const items: MenuItem[] = [
     to: '/info/[side]',
     as: '/info/for-bedrifter',
   },
+  { key: 'galleri', name: 'galleri', to: '/galleri' },
```
Comment (ivarnakken) [inline components/Header/index.tsx:52]: Please capitalize the first letter 🙏🏼

File: pages/galleri/index.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+// for some fucking reason
+const images = [
+  'DSC_1043.jpg',
+  'DSC_1003.jpg',
+  'DSC_1053.jpg',
+  'DSC_1105.jpg',
+  'DSC_1128.jpg',
+  'DSC_1137.jpg',
+  'DSC_1141.jpg',
+  'DSC_1145.jpg',
```
Comment (ivarnakken) [inline pages/galleri/index.tsx:17]: Is the plan to hard code all images in here?

**Review by ivarnakken — COMMENTED** [review]
Cool shit

Kinda skeptical of hard coding the images but okay

File: pages/galleri/index.tsx (hunk 2 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:17]: I talked it over with Ludvig and we concluded that implementing a smarter solution would be difficult and could be done in a later pr. 

But it is totally in my plans to do so.

File: components/Header/index.tsx (hunk 1 above)
Comment (Arashfa0301) [reply components/Header/index.tsx]: Right on.

File: pages/galleri/index.tsx (hunk 2 above)
Comment (LudvigHz) [reply pages/galleri/index.tsx]: On another note, the comment here doesn't really make sense?

File: pages/galleri/index.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  // Close Modal
+  const handleCloseModal = () => {
+    setOpenModal(false);
+  };
+
+  // Previous Image
+  const prevSlide = () => {
+    slideNumber === 0
+      ? setSlideNumber(images.length - 1)
+      : setSlideNumber(slideNumber - 1);
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:117]: A cool, but maybe more complicated way to do this is:

```ts
(images.length - 1 + slideNumber -1) % images.length -1
```

If you ever want to save precious CPU cycles :))

File: pages/galleri/index.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    name: string;
+    top?: string;
+    right?: string;
+    left?: string;
+    onClick: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  }) => {
+    return (
+      <>
+        {/* eslint-disable-next-line*/}
+        {/* @ts-ignore*/}
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:143]: Please don't do this. Fix the types instead

File: pages/galleri/index.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        handleCloseModal();
+      }
+      if (event.key === 'ArrowRight') {
+        nextSlide();
+      }
+      if (event.key === 'ArrowLeft') {
+        prevSlide();
+      }
+    },
+    [slideNumber]
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:172]: Why is this a dep here?

File: pages/galleri/styles.module.css [hunk 6]
```diff
@@ -0,0 +1,14 @@
+.gallery {
+  width: 230px;
+  margin: 8px 20px 0;
+  display: -ms-grid;
```
Comment (LudvigHz) [inline pages/galleri/styles.module.css:4]: Do you need this?

Same with the other -ms properties below

File: pages/galleri/styles.module.css [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  display: grid;
+  grid-gap: 4px 4px;
+  -ms-grid-columns: 1fr 1fr 1fr;
+  -ms-grid-rows: 2;
+  grid-template-columns: 1fr 1fr 1fr;
+  border: 1px solid transparent;
```
Comment (LudvigHz) [inline pages/galleri/styles.module.css:10]: Transparent border?

File: pages/galleri/index.tsx [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+                height={330}
+              />
+            </FullScreenImageDiv>
+          </SliderWrap>
+        )}
+
+        <Title>Galleri</Title>
+        <GalleryWrap>
+          {images.map((slide, index) => {
+            return (
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:212]: You can drop the return and have it be implicit here

File: pages/galleri/index.tsx [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+    },
+    [slideNumber]
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', changeImageOnKey);
+    return function cleanup() {
+      document.removeEventListener('keydown', changeImageOnKey);
+    };
+  });
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:180]: Hmmm. Are you sure this is the best way to do this?

File: pages/galleri/index.tsx [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  gap: 10px;
+  align-items: center;
+  justify-content: center;
+  max-width: 1120px;
+  margin: 0 auto 60px;
+`;
+
+const SliderWrap = styled('div')`
```
Comment (LudvigHz) [inline pages/galleri/index.tsx:28]: Just a note that react-modal is a thing that exists.

File: pages/galleri/index.tsx (hunk 2 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx]: Already removed. Really unprofessional of me to curse 🥲

File: pages/galleri/index.tsx (hunk 3 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:117]: It works but I am struggling to understand how. Could you explain @LudvigHz

File: pages/galleri/styles.module.css (hunk 6 above)
Comment (Arashfa0301) [reply pages/galleri/styles.module.css:4]: This file is unnecessary.

File: pages/galleri/styles.module.css (hunk 7 above)
Comment (Arashfa0301) [reply pages/galleri/styles.module.css:10]: This file is unnecessary.

File: pages/galleri/index.tsx (hunk 8 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:212]: Yes. Slipped my mind.

File: pages/galleri/index.tsx (hunk 10 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:28]: Noted. This work just fine though

File: pages/galleri/index.tsx (hunk 5 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:172]: Doesn't need to be :)

File: pages/galleri/index.tsx (hunk 9 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:180]: Only way I found out.

File: pages/galleri/index.tsx (hunk 4 above)
Comment (Arashfa0301) [reply pages/galleri/index.tsx:143]: Its impossible (Really difficult) with io-icons. We do it the same way I Lego.

**Pull request merged**